File: upgrade.js
/* eslint-disable @typescript-eslint/no-var-requires */
const version = require('./version.json');
const fts = require('./fts-version.json');
const { exec } = require('child_process');

const latest_fts = fts[fts.length - 1];
const IMAGE_REPOSITORY = 'hotrungnhan/postgres';
Object.entries(version).forEach(([key, value]) => {
  //   const platform = `--platform ${value.platform.join(',')}`;
  const platform = '';
  const tag = value.tag
    .concat([`${key}_${latest_fts.version}`])
    .map((t) => `-t ${IMAGE_REPOSITORY}:${t}`)
    .join(' ');

  const env = { FTS_VERSION: latest_fts.version, ROOT_IMAGE_TAG: key };
  exec(
    `docker buildx build ${platform} ${tag} --push .`,
    { env: env, shell: 'bash', windowsHide: false },
    (err, stdout, stderr) => {
      if (err) throw err;
      if (stderr) throw stderr;
      console.log(stdout);
      exec(`docker push ${tag}}`);
    },
  );
});

File: upgrade.py
import json

def read_json(filepath):
    with open(filepath, 'r') as f:
        return json.load(f)


print(f"echo '🏃 running builder'")
### environent
IMAGE_REPOSITORY = 'hotrungnhan/postgres'
FTS_VERSION=read_json("./fts-version.json")[-1]["version"]

for [key,value] in read_json('./version.json').items():
    # platform param
    platform = f"--platform {','.join(value['platform'])}"
    # tag param
    tags = value["tag"] + [f"{key}-{FTS_VERSION}"]
    tags= map(lambda t:f"-t {IMAGE_REPOSITORY}:{t}",tags)
    tags= " ".join(tags)

    print(f"docker buildx build --build-arg POSTGRES_VERSION={key} --build-arg FTS_VERSION={FTS_VERSION} {platform} {tags} --push .")
print(f"echo '✅ done check it out at: https://hub.docker.com/repository/docker/{IMAGE_REPOSITORY}'")
